src/List.ls

```diff
@@ -291,13 +291,13 @@ zip-with = (f, xs, ys) -->
   result
 
 zip-all = (...xss) ->
-  min-length = 9e9
+  min-length = undefined
```

Comment:
@joneshf Doesn't that give the maximum length, which would change the behavior? I have to admit I do have some problems wrapping my head around the `<?=` / `>?=`, but the current implementation seems to give the minimum length, so I'd guess `>?=` would give the maximum.

@michaelficarra Yes, this would be one way. However, IMHO `undefined` is more readable and doesn't make this a corner case. The comparison operator behaviors on `undefined` (etc) are sort of designed for this purpose, ie so that they "fall through" nicely for most purposes without extra checking (a bit like Maybe monads from functional perspective). This is discussed more in context of IEEE-754 NaN's, which is almost an identical case: http://stackoverflow.com/questions/1565164/what-is-the-rationale-for-all-comparisons-returning-false-for-ieee754-nan-values

I think the best solution would be

    min-length = Math.min ...map (.length) xss

Although there are two problems:
1. It may be slower and/or use more memory as map doesn't produce iterator, but a list. The interpreter/JIT may or may not be smart enough to optimize the temporary list away. Probably isn't.
1. `Math.min` returns `Infinity` on empty list, so it would need a separate check, or we'd get an actual infinite loop. Returning `Infinity` sort of makes sense, however, I think returning `undefined` would be better. But I guess we have to live with JS stupidities.

But anyway, I don't want to invest much time on such a trivial fix, and I don't really care how it's done as long as the behavior is correct, so please no bikeshedding. If somebody comes up with a cleaner solution, I don't mind this PR being rejected instead of a better one. However, I'd like to see this fixed quite fast, because otherwise I have to hack the checks on call sites.